File: android/app/src/main/java/com/costoda/dittoedgestudio/ui/mainstudio/profile/ProfileFooterStrip.kt
package com.costoda.dittoedgestudio.ui.mainstudio.profile

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.text.selection.SelectionContainer
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import com.costoda.dittoedgestudio.domain.model.QueryProfile

/**
 * Subtle footer below the plan: profile UUID, app ID, state — matching the VS Code
 * profile page's trailing strip. Selectable so a user copying these into a bug
 * report doesn't need to retype the GUIDs.
 */
@Composable
fun ProfileFooterStrip(
    profile: QueryProfile,
    modifier: Modifier = Modifier,
) {
    SelectionContainer {
        Text(
            text = "profile: ${profile.id.ifEmpty { "—" }}" +
                " · db: ${profile.appId.ifEmpty { "—" }}" +
                " · state: ${profile.state.ifEmpty { "—" }}",
            style = MaterialTheme.typography.labelSmall,
            fontFamily = FontFamily.Monospace,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = modifier.fillMaxWidth().testTag("ProfileFooter"),
        )
    }
}

File: SwiftUI/EdgeStudio/Components/ProfileViewer/ProfileFooterStrip.swift
import SwiftUI

/// Subtle footer below the operator cards: profile UUID, app ID,
/// state. Mirrors the reference at `screens/profile-viewer.png`
/// where the same trailing strip appears as small grey text.
///
/// Selectable so a user copying these into a bug report doesn't need
/// to retype the GUIDs.
struct ProfileFooterStrip: View {
    let profile: QueryProfile

    var body: some View {
        HStack(spacing: 8) {
            field(label: "profile", value: profile.id)
            divider
            field(label: "db", value: profile.appId)
            divider
            field(label: "state", value: profile.state)
            Spacer(minLength: 0)
        }
        .font(.caption2.monospaced())
        .foregroundStyle(.secondary)
        .textSelection(.enabled)
    }

    private var divider: some View {
        Text("·").foregroundStyle(.tertiary)
    }

    private func field(label: String, value: String) -> some View {
        HStack(spacing: 4) {
            Text("\(label):")
                .foregroundStyle(.tertiary)
            Text(value.isEmpty ? "—" : value)
        }
    }
}
